spunky-sputniks/src/discobase/database.py

```diff
@@ -0,0 +1,459 @@
+                break
+
+        if not found_guild:
+            logger.info("No guild found, making one.")
+            self.guild = await self.bot.create_guild(name=self.name)
+        else:
+            logger.info("Found an existing guild.")
+            self.guild = found_guild
+
+        # Unlock database, but don't wakeup the user.
+        self._internal_setup_event.set()
+        await self.build_tables()
+        # At this point, the database is marked as "ready" to the user.
+        self._setup_event.set()
+
+        assert self._metadata_channel is not None
+        logger.info(
+            f"Invite to server: {await self._metadata_channel.create_invite()}"
+        )
+        logger.info("Syncing slash commands, this might take a minute.")
+        logger.debug(f"Synced slash commands: {await self.bot.tree.sync()}")
+
+    async def build_tables(self) -> None:
+        """
+        Generate all internal metadata and construct tables.
+
+        Calling this manually is useful if e.g. you want
+        to load tables *after* calling `login` (or `login_task`, or
+        `conn`, same difference.)
+
+        This method is safe to call multiple times.
+
+        Example:
+            ```py
+            import asyncio
+            import discobase
+
+            async def main():
+                db = discobase.Database("My database")
+                db.login_task("My bot token")
+
+                @db.table
+                class MyLateTable(discobase.Table):
+                    something: str
+
+                # Load MyLateTable into database
+                await db.build_tables()
+
+            asyncio.run(main())
+            ```
+        """
+        if not self.guild:
+            self._not_connected()
+
+        self._metadata_channel = await self._metadata_init()
+        tasks = [
+            asyncio.ensure_future(
+                TableCursor.create_table(
+                    table,
+                    self._metadata_channel,
+                    self.guild,
+                )
+            )
+            for table in self.tables.values()
+        ]
+        logger.debug(f"Creating tables with gather(): {tasks}")
+        await asyncio.gather(*tasks)
+
+    async def wait_ready(self) -> None:
+        """
+        Wait until the database is ready.
+        """
+        logger.info("Waiting until the database is ready.")
+        await self._setup_event.wait()
+        logger.info("Done waiting!")
+        # See #49, we need to propagate errors in `on_ready` here.
+        if self._on_ready_exc:
+            logger.error("on_ready() failed, propagating now.")
+            raise self._on_ready_exc
+
+    def _find_channel(self, cid: int) -> discord.TextChannel:
+        # TODO: Implement caching for this function.
+        if not self.guild:
+            self._not_connected()
+
+        index_channel = [
+            channel for channel in self.guild.channels if channel.id == cid
+        ][0]
```

Comment:
```suggestion
        index_channel = self.guild.get_channel(cid)
```
This will also save you from the exception you'll get if there's no channel matching `cid`.